pkg/engine/virtualmachines.go

```diff
@@ -498,8 +498,14 @@ func createAgentAvailabilitySetVM(cs *api.ContainerService, profile *api.AgentPo
 	storageProfile := compute.StorageProfile{}
 
 	if profile.IsWindows() {
-		imageRef := profile.ImageRef
+		// Priority:
+		//   1. ImageRef in agent pool
+		//   2. Custom image in WindowsProfile. AKS does not use this for now but it may use this later for testing.
+		//   3. ImageRef in WindowsProfile
+		//   4. PIR image in WindowsProfile
+		windowsProfile := cs.Properties.WindowsProfile
```

Comment:
the rule is a bit complicated, why do we need to use both agent pool and windows profile? can we not combine them into one?

Reply:
Current code is using ImageRef in WindowsProfile. To support more Windows image, for example, 2004, we need to support multiple VHD images in the same cluster so we need to use ImageRef (SIG uses) in agent pool. Later the VHD info in WindowsProfile will be deprecated.